packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -100,7 +100,17 @@ class DriveCommand extends RunCommandBase {
 
     if (!argResults['use-existing-app']) {
       printStatus('Starting application: ${argResults["target"]}');
-      int result = await appStarter(this, getBuildMode());
+
+      if (getBuildMode() == BuildMode.release) {
+        printError(
+          'Flutter Driver does not support running in release mode. '
```

Comment:
As-is is fine, but another phrasing: "Flutter Drive requires Observatory which is not available in --release mode, please use --profile".